python_modules/dagster/dagster/core/asset_defs/assets_job.py

```diff
@@ -111,14 +114,20 @@ def asset2(asset1):
     )
 
     all_resource_defs = dict(resource_defs)
-    for asset_def in assets:
-        for resource_key, resource_def in asset_def.resource_defs.items():
+    for assets_def in assets:
+        # Until we remove the ability to specify resource defs externally here, cannot check all asset defs for satisfying their resource requirements.
```

Comment:
Thoughts on using "`with_resources`" here on the assets so that the code paths afterwards don't need to account for the resource_defs that are passed in?

Reply:
cool idea. `with_resources` just landed so will add that.